### PR TITLE
Cleanup for new password reset rules.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request includes follow-up from #1051:

🍃 Removes an old validation class from our `$dontReport` array that no longer exists in this Laravel release. 47b2d7b

🎋 Write validation errors to the application log. This will help us keep an eye on how often these occur. d8be3c0

*️⃣ Updates the placeholder for the registration form's password field to specify new length requirement. 08a7bad

⛑️ Applies new password validation rules to reset form as well. aa6a19b

### How should this be reviewed?

I'd recommend reviewing commit-by-commit!

### Any background context you want to provide?

I've got the changes to separate out the "change password" form staged, but holding for another PR to keep this small.

### Relevant tickets

References [Pivotal #173718892](https://www.pivotaltracker.com/story/show/173718892).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [ ] If new attributes were added to users, there is a corresponding PR to surface these in Aurora.
- [ ] If new attributes were added to users, then the data team already knows about these changes.

### CODE REVIEW

File: app/Http/Controllers/Web/ResetPasswordController.php [hunk 1]
```diff
@@ -35,6 +36,20 @@ public function __construct()
...
+     */
+    protected function rules()
+    {
+        return [
+            'token' => 'required',
+            'email' => 'required|email',
+            'password' => PasswordRules::changePassword(request()->email),
+        ];
+    }
+
```
Comment (mendelB) [inline app/Http/Controllers/Web/ResetPasswordController.php:52]: Where do these rules actually get applied -- is there internal functionality that pulls in the `rules` which is now just customized for this controller?

**Review by mendelB — APPROVED** [review]
🎱

File: app/Http/Controllers/Web/ResetPasswordController.php (hunk 1 above)
Comment (DFurnes) [reply app/Http/Controllers/Web/ResetPasswordController.php:52]: Yep! Most of the logic for this controller comes from the `ResetsPasswords` trait, which uses these rules [here](https://github.com/laravel/framework/blob/d2e513a608c34fcea668bdabe9f38c8ed0b335d4/src/Illuminate/Foundation/Auth/ResetsPasswords.php#L40).

Comment (DFurnes) [reply app/Http/Controllers/Web/ResetPasswordController.php:52]: It does make me wonder if this is a convention we'd want to adopt for our own shared validation rules in controllers. We tend to store these on a `$this->rules` property, if anything...

Comment (mendelB) [reply app/Http/Controllers/Web/ResetPasswordController.php:52]: ohhh cool -- I dig it!

**Pull request merged**